nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -677,20 +720,37 @@ def setup_streaming_params(
 
         self.streaming_cfg = streaming_cfg
 
-    def get_initial_cache_state(self, batch_size=1, dtype=torch.float32, device=None):
+    def get_initial_cache_state(self, batch_size=1, dtype=torch.float32, device=None, random_init=False):
         if device is None:
             device = next(self.parameters()).device
+        if random_init:
+            create_tensor = torch.randn
+        else:
+            create_tensor = torch.zeros
         last_time_cache_size = self.conv_context_size[0]
-        cache_last_channel = torch.zeros(
-            (self.streaming_cfg.last_channel_num, batch_size, 0, self.d_model), device=device, dtype=dtype
+        cache_last_channel = create_tensor(
+            (
+                batch_size,
+                self.streaming_cfg.last_channel_num,
+                self.streaming_cfg.last_channel_cache_size,
+                self.d_model,
+            ),
+            device=device,
+            dtype=dtype,
         )
-        cache_last_time = torch.zeros(
-            (self.streaming_cfg.last_time_num, batch_size, self.d_model, last_time_cache_size),
+        cache_last_time = create_tensor(
```

Comment:
Why batch size has moved to the first dimension?

Reply:
This diff does not reflect the latest changes. It's back where it was.